docs/software/container-development-guide.md

```diff
@@ -0,0 +1,107 @@
+   Update the `docker_repos` in `build_docker.sh` to the name of the Docker Hub repo
+1. Set the `+x` bit on `travis/build_docker.sh`
+1. Send the `opensciencegrid` GitHub repoository URL to the Software Manager and ask them to do the following:
+    1. Enable the repository in Travis-CI
+    1. Add the OSG Software Docker Hub account credentials to the Travis CI repository as secure variables
+       `DOCKER_USERNAME` and `DOCKER_PASSWORD`.
+       Ensure that `Display value in build log` remains unset.
+       Escape any special characters with `\`.
+    1. Enable weekly cron builds from `master` and set `Always run`
+
+Triggering Container Image Builds
+---------------------------------
+
+To build a new version of an [existing container image](#creating-a-new-container-image), e.g. for a new RPM version of
+software in the container, you can kick off a new build in one of two ways:
+
+- **If there are no changes necessary to the container packaging:** go to the repository's latest Travis-CI build off of
+  `master`, e.g. <https://travis-ci.org/opensciencegrid/docker-xcache>, and click "Restart build"
+- **If changes need to be made to the container packaging:** submit a pull request with your changes to the relevant
+  GitHub repository and request that another team member review it.
+  Once merged into `master`, a Travis-CI build should start automatically.
+
+If the Travis-CI build completes successfully, you should shortly see new `fresh` and timestamp tags appear in the
+DockerHub repository.
+
+!!! note "Automatic weekly rebuilds"
+    If the repo's Travis-CI is configured as above, container images will automatically rebuilds, and therefore pick up
+    new packages available in minefield once per week.
+
+Managing Tags in DockerHub
+--------------------------
+
+### Adding tags ###
+
+Images that have passed acceptance testing should be tagged as `stable`:
+
+1. Install the `jq` utility:
+
+        yum install jq
+        
+1. Get the SHA256 repo digest of the image that the user has tested
+1. Go to the Docker Hub repo (e.g., <https://hub.docker.com/r/opensciencegrid/frontier-squid/tags>) and find the
+   `<TIMESTAMP TAG>` (e.g., `20191118-1706`) corresponding to the digest in the previous step
+1. Add your Docker Hub user/pass to a file with `600` permissions:
+
+        export user=<dockerhub username>
+        export pass=<dockerhub password>
+
+1. Run the Docker container image tagging command from [release-tools](https://github.com/opensciencegrid/release-tools/):
+
+        ./dockerhub-tag-fresh-to-stable.sh <IMAGE NAME> <TIMESTAMP TAG>
+
+1. Clean up your Docker Hub user/pass file
+
+### Removing tags ###
+
+Via the web interface, requires write permissions
+
+Making Slim Containers
+----------------------
+
+Here are some resources for creating slim, efficient containers:
+
+- <https://github.com/opensciencegrid/topology/pull/399>
+- <https://docs.docker.com/develop/develop-images/multistage-build/>
```

Comment:
I don't think you need three links for multi-stage builds. It's cool but it's really only good for when stage 1 is compiling your software and stage 2 is installing it. Most of our containers are 'yum install stuff from the osg repos and do some config' which doesn't benefit from multi-stage builds.

This link might be useful: https://developers.redhat.com/blog/2016/03/09/more-about-docker-images-size/